code/modules/station_goals/bsa.dm

```diff
@@ -178,9 +178,11 @@
 
 /obj/machinery/bsa/full/proc/fire(mob/user, turf/bullseye)
 	var/turf/point = get_front_turf()
+	var/obj/effect/temp_visual/bsa_splash/wave
 	for(var/turf/T in getline(get_step(point,dir),get_target_turf()))
 		T.ex_act(EXPLODE_DEVASTATE)
 	point.Beam(get_target_turf(),icon_state="bsa_beam",time=50,maxdistance = world.maxx) //ZZZAP
+	new wave (point, dir)
```

Comment:
This doesn't read like it would work. Have you tested it?

I think you want:
```dm
new /obj/effect/temp_visual/bsa_splash(point, dir)
```

Reply:
I think I missed this on the first pass, it was saved in my backup of the changed files and got thrown in. Your version's working as intended.